src/chainClient.ts

```diff
@@ -0,0 +1,98 @@
+import "@polkadot/api-augment";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { WalletClient, BalanceInfo } from "./walletClient";
+
+
+import { BN, formatBalance } from '@polkadot/util';
+
+class PolkadotClient implements WalletClient {
+  walletAddress: string;
+
+  constructor(walletAddress: string) {
+    this.walletAddress = walletAddress;
+  }
+
+  public async getBalance(): Promise<BalanceInfo> {
+    const wsProvider = new WsProvider("wss://rpc.polkadot.io");
+    const api = await ApiPromise.create({ provider: wsProvider });
+
+    let {
+      data: { free: previousFree },
+      nonce: previousNonce,
+    } = await api.query.system.account(this.walletAddress);
+
+
+
+
+    const chainDecimals = api.registry.chainDecimals[0];
```

Comment:
probably comment on why you are hardcoding the 0th index on chainDecimals, and what a chainDecimal is